tests/src/common/test_case.rs

```diff
@@ -414,12 +470,26 @@ where
 
         self.message_receiver.assert_empty();
 
-        self.dispatcher_addr = Some(dispatcher_addr);
-
         self
     }
+}
+
+impl<Lpn> Deref for TestCase<Lpn>
+where
+    Lpn: Currency,
+{
+    type Target = GenericTestCase;
+
+    fn deref(&self) -> &Self::Target {
+        &self.test_case
+    }
+}
 
-    pub fn leaser(&self) -> Addr {
-        self.leaser_addr.clone().unwrap()
+impl<Lpn> DerefMut for TestCase<Lpn>
+where
+    Lpn: Currency,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.test_case
     }
 }
```

Comment:
please refrain from misusing these two traits ... we have already spoken about that. It's preferable to use explicit indirections when doing calls.